lib/nameMatcher.js

```diff
@@ -0,0 +1,307 @@
+'use strict';
+/**
+ * nameMatcher module
+ * exports nameMatcher object to expose methods for adding names and matching against them.
+ */
+
+
+// Private variables and external modules
+var rsvp = require('rsvp');
+// Set up rsvp global error handler to handle uncaught errors thrown within a promise-chain
+rsvp.on('error', function catchGlobalRSVPError(err) {
+    console.log('uncaught rejection of promise', err);
+    if (err instanceof Error && err.stack) {
+        console.error(err.stack);
+    }
+});
+var Promise = rsvp.Promise;
+var redis = require('redis');
+var natural = require('natural');
+var _ = require('underscore');
+
+var nameMatcher = {};
+var redisClient;
+// firstNames = e.g. {bob: bobMetaphoneHash, rachel: rachelMetaphoneHash}
+var firstNames;
+var lastNames;
+var firstNamesDev;
+var lastNamesDev;
+
+
+//static private helper functions
+/**
+ * Helper function to wrap standard callback-based async functions in a promise
+ * @param fn {function} The function to be called
+ * @param args {array} array of arguments to be passed to the function
+ * @param _this {object} (optional) object to use as `this` when calling the function defaults to null
+ * @return {Promise} promise to be resolved or rejected when the async fn is complete
+ */
+var promisify = function (fn, args, _this) {
+    _this = _this || null;
+    return new Promise(function promisifyWrapper(resolve, reject) {
+        var promiseCallback = function (err, result) {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(result);
+            }
+            return;
+        };
+        args.push(promiseCallback);
```

Comment:
if someone ever passes in the actual `arguments` object, this will break.  you can do a
```js
if (_.isArguments(args)) {
    args = _.toArray(args);
}
```
or similair